src/Providers/AddonServiceProvider.php

```diff
@@ -3,40 +3,145 @@
 namespace Statamic\Providers;
 
 use Closure;
+use Illuminate\Console\Command;
 use Illuminate\Console\Scheduling\Schedule;
+use Illuminate\Foundation\Events\Dispatchable;
 use Illuminate\Support\Facades\Event;
 use Illuminate\Support\Facades\Gate;
 use Illuminate\Support\Facades\Route;
 use Illuminate\Support\ServiceProvider;
+use Statamic\Actions\Action;
 use Statamic\Exceptions\NotBootedException;
 use Statamic\Extend\Manifest;
 use Statamic\Facades\Addon;
+use Statamic\Fields\Fieldtype;
+use Statamic\Modifiers\Modifier;
+use Statamic\Query\Scopes\Scope;
 use Statamic\Statamic;
 use Statamic\Support\Str;
+use Statamic\Tags\Tags;
+use Statamic\UpdateScripts\UpdateScript;
+use Statamic\Widgets\Widget;
 
 abstract class AddonServiceProvider extends ServiceProvider
 {
+    /**
+     * Array of event class => Listener class
+     * @var array<class-string<Dispatchable>, class-string[]>
+     */
     protected $listen = [];
+
+    /**
+     * @var list<class-string>
```

Comment:
Why do you use `list` instead of `array`?

I'm not saying it's wrong, I just want to understand.

Reply:
In PHP `array` can be multiple things as it combines what other languages call a map and an array into one type. For that reason static analyzers (+ PHPStorm and others) have invented the `list` type to differentiate between an associative array and a sequential one.

* `array<TKey, TValue>` is associative
* `TValue[]` is basically a shortcut for `array<int, TValue>`
* `list<TValue>` additionally enforces sequential integer arrays. It's not really important in these instances, but I like to add as much type detail as possible

Reply:
I see no mention of it on the phpstan website. 🤔

Reply:
You are correct - PHPStan does not document it. Here is a link to Psalms documentation about it: https://psalm.dev/docs/annotating_code/type_syntax/array_types/

While PHPStand does not document it, it does interpret `list<TValue>` as `array<int, TValue>`, which is still more lenient than what Psalm does, but it does support the type.

For reference:
* `array_filter` being a prime example, why I usually like using `list<TValue>` (with Psalm): https://psalm.dev/r/2a35e60e36
* The same code not yielding issues in PHPStan, but the integer being detected as incorrect: https://phpstan.org/r/a557359f-ba17-4b8a-8ef9-6c0c6930dc02

Reply:
Cool cool, thanks!